vault/auth.go

```diff
@@ -156,8 +152,8 @@ func (c *Core) enableCredentialInternal(ctx context.Context, entry *MountEntry,
 
 	// Check for the correct backend type
 	backendType := backend.Type()
-	if entry.Type == "plugin" && backendType != logical.TypeCredential {
-		return fmt.Errorf("cannot mount %q of type %q as an auth backend", entry.Config.PluginName, backendType)
+	if backendType != logical.TypeCredential && entry.Type != "token" {
```

Comment:
Instead of hard-coding exceptions like these, I could also set the backend type to `logical.TypeCredential` on the `"token"` backend, and do the same with similar hard-coded exceptions. Would love to get feedback on that.

Reply:
I don't know the impact of the change, but FWIW I also ended up having hard-coded exceptions for "unknown" backend types in the OpenAPI stuff. If we add types, I can remove that handling as well.

Reply:
:heavy_check_mark: